### ISSUE TITLE
Should logging include homepage/root URLs?

### ISSUE THREAD

**Issue by petecooper** [textpattern/textpattern#1264]
### Expected behaviour

Homepage hits appear in Logging as a `/` or similar.

### Actual behaviour

Homepage hits appear in Logging as an empty cell.

### Steps to reproduce

* Enable all Logging
* Visit various pages and URLs within the site
* Note nothing is recorded for the URL in Page column in Visitor Logs

#### Additional information

Empty cells in the following screenshot represent homepage/root/`default` hits:

![safariscreensnapz001](https://user-images.githubusercontent.com/226578/39871991-7eae852e-545e-11e8-8319-931896a597ad.png)

**Comment by petecooper** [comment]
(Follow up – I don't use internal logging at all, so this might be how it's always been, please instaclose if that's the case.)

**Comment by philwareham** [comment]
Yeah, I noticed that too a couple of days back (don't use logging myself - just turned it on for GDPR testing). I should I think show something, looks weird without it - just don't know what. Not sure if forward slash would work (probably would have to append it to the child URLs too)?

**Comment by Bloke** [comment]
Hmm, yeah, that is odd. We could use the URL for the "home page" but that might then look odd alongside all the other pages that don't have it. And putting it on all pages just clutters the panel.

Maybe just use something we have a `[common]` gTxt string for, and put it in square brackets or summat. Like:

[Home]

or

[Site]

??

**Comment by philwareham** [comment]
Changing line 262 of `txp_log.php` from:

    $log_page = href(txpspecialchars($log_anchor), txpspecialchars($log_page), ' target="_blank"');

To:

    $log_page = href('/'.txpspecialchars($log_anchor), txpspecialchars($log_page), ' target="_blank"');

Adds a preceding `/` to all entires which *may* fix this. What I don't know is if URLs are relative to the domain root or to the root of the Textpattern location?

**Comment by Bloke** [comment]
That works and I can't see a downside: it's only the visible part of the link that's being modified so shouldn't affect the destination in any way. We'll have to get someone to test this in a multi-site setup to see what happens there. @jools-r ?

**Comment by jools-r** [comment]
Almost works. The visible part of the link acquires a preceding slash, so the home page is now `/`, but the href of the link itself is not relative to the front side link. It needs to be:

```
$log_page = href('/'.txpspecialchars($log_anchor), rtrim(hu,'/').txpspecialchars($log_page), ' target="_blank"');
```
with the trailing slash trimmed off `hu` (or alternatively ltrim the preceding slash off `$log_page`). I think that should be good for non-multisite sites too?

I think the `/` as an indicator of the homepage is fine.

**Comment by jools-r** [comment]
Just confirming that the above line works in multisite and regular non-multisite configurations.

**Comment by Bloke** [comment]
Fixed in 66e67e44. Sorry, Jools, didn't see your PR until after I'd hacked the code.

**Comment by jools-r** [comment]
No problem. I wasn't sure if you were waiting for me. Thanks for updating one way or the other...
